### ISSUE TITLE
Deployment fails

### ISSUE THREAD

**Issue by Ingo-FP-Angel** [WeAreFairphone/fpangels-hp#10]
The last CI builds at https://travis-ci.org/WeAreFairphone/fpangels-hp/builds all failed.

From the latest log https://travis-ci.org/WeAreFairphone/fpangels-hp/builds/625381150 in the "Deploying application" step:

```
Couldn't push the build to github.com/WeAreFairphone/fpangels-hp.git:gh-pages
```

**Comment by StefanBrand** [comment]
The repositories are gone from my TravisCI account. Thus I cannot make any modifications anymore. 😬 Can you work with organizations' repos now?

**Comment by Ingo-FP-Angel** [comment]
No, my TravisCI account has no access to any organization's repos :-(

**Comment by Ingo-FP-Angel** [comment]
There is probably one more useful piece of information, taken from https://travis-ci.org/WeAreFairphone/fpangels-hp/requests

```
GitHub Pages branch not included via configuration.
```

**Comment by StefanBrand** [comment]
Maybe you must add a new environment variable, as indicated here: (This was probably connected to my account.) :)

https://github.com/WeAreFairphone/fprsmap/blob/2116fde0dec4a518e4665b8e23fd2e70410d900c/.travis.yml#L11

**Comment by StefanBrand** [comment]
@ff2u here: https://github.com/WeAreFairphone/meta/issues/2#issuecomment-566288962
> I assume this is - since 4 days - a problem with the secure token (GITHUB_TOKEN) to access the according two WeAreFairphone repositories. But I can't check or try to fix that as I assume I don't have the necessary rights (I can't access the settings of the repositories).

Yep, would say so too. The `GITHUB_TOKEN` is probably bound to my account. 😬 One of the admins might have to set a new one: @Roboe @z3ntu @snevas (?)

**Comment by z3ntu** [comment]
So if I understand the deployment of this repo correctly, travis-ci builds every new commit of master and then pushes the result to the gh-pages branch which is taken by github and displayed on the wearefairphone.github.io domain. And travis also auto-runs every 24 hours to regenerate the site (with new events I guess)

Now that I know what this is about, it would be good to use deploy keys for deploying the results from travis - see https://docs.travis-ci.com/user/deployment-v2/providers/pages/

I don't really understand the docs how this should be set up, but that way is the best I think.

**Comment by ff2u** [comment]
Well, I think at first a personal access token has to be created in github following
https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line
I can't do it as I don't have the necessary rights in WeAreFairphone. 
Then this token has to be configured in Travis following
https://docs.travis-ci.com/user/environment-variables#defining-variables-in-repository-settings
I assume that's it then.

**Comment by Roboe** [comment]
Fixed. The new personal access token was generated with `public_repo` permissions, for reference.
Both repos have been re-built and deployed successfully.

**Comment by StefanBrand** [comment]
Thank you! :hugging: 
-- 
Diese Nachricht wurde von meinem Fairphone mit K-9 Mail gesendet.

Am 23. Dezember 2019 02:54:19 MEZ schrieb Roberto MF <notifications@github.com>:
>Closed #10.

**Comment by Roboe** [comment]
Ah, I forgot to say: I don't like the personal access token method, security-wise. I think @z3ntu has the same opinion, since he mentioned the deploy SSH keys method &mdash;which is bound to the _repo_, not the _user_.

I'll try to understand how this method works in the next days. I know a private key needs to be encrypted with a `travis` tool and commited to the repo, but I don't have all the details just yet. For example: I don't think we need a [build script](https://medium.com/@simon.legner/deploy-to-github-pages-using-travis-ci-and-deploy-keys-db84fed7a929) nowadays, but [Travis docs](https://docs.travis-ci.com/user/deployment-v2/providers/pages/) are quite scarce at that respect.
